src/Plugins/RpcServer/RpcServer.cs

```diff
@@ -258,27 +262,87 @@ public async Task ProcessAsync(HttpContext context)
             await context.Response.WriteAsync(response.ToString(), Encoding.UTF8);
         }
 
+
         private async Task<JObject> ProcessRequestAsync(HttpContext context, JObject request)
         {
             if (!request.ContainsProperty("id")) return null;
-            JToken @params = request["params"] ?? new JArray();
+            var @params = request["params"] ?? new JArray();
             if (!request.ContainsProperty("method") || @params is not JArray)
             {
                 return CreateErrorResponse(request["id"], RpcError.InvalidRequest);
             }
-            JObject response = CreateResponse(request["id"]);
+
+            var jsonParameters = (JArray)@params;
+            var response = CreateResponse(request["id"]);
             try
             {
-                string method = request["method"].AsString();
+                var method = request["method"].AsString();
                 (CheckAuth(context) && !settings.DisabledMethods.Contains(method)).True_Or(RpcError.AccessDenied);
-                methods.TryGetValue(method, out var func).True_Or(RpcErrorFactory.MethodNotFound(method));
-                response["result"] = func((JArray)@params) switch
+
+                if (methods.TryGetValue(method, out var func))
+                {
+                    response["result"] = func(jsonParameters) switch
+                    {
+                        JToken result => result,
+                        Task<JToken> task => await task,
+                        _ => throw new NotSupportedException()
+                    };
+                    return response;
+                }
+
+                if (_methodsWithParams.TryGetValue(method, out var func2))
                 {
-                    JToken result => result,
-                    Task<JToken> task => await task,
-                    _ => throw new NotSupportedException()
-                };
-                return response;
+                    var paramInfos = func2.Method.GetParameters();
+                    var args = new object[paramInfos.Length];
+
+                    for (var i = 0; i < paramInfos.Length; i++)
+                    {
+                        var param = paramInfos[i];
+                        if (jsonParameters.Count > i && jsonParameters[i] != null)
+                        {
+                            try
+                            {
+                                args[i] = ConvertParameter(jsonParameters[i], param.ParameterType);
+                            }
+                            catch (Exception e)
+                            {
+                                if (param.IsOptional)
```

Comment:
It could be optional, but if you send the argument in wrong format you should not get the default value

Reply:
If a value is applied, will not adopt the optional value anymore, updated the logic to throw exception directly.